plugin/keyring_vault/keyring_vault-test/keyring_vault-test.cc

```diff
@@ -35,19 +59,25 @@ static void* generate(void *arg)
     sprintf(key_id, "Key#%d", key_nr);
     strcpy(key_type, "AES");
     sprintf(user, "User#%d", key_nr);
+    Key_in_keyring key_in_keyring(key_id, user);
 
     my_bool result= FALSE;
 
     if((result= !mysql_key_generate(reinterpret_cast<const char*>(key_id),
                                     reinterpret_cast<const char*>(key_type),
                                     reinterpret_cast<const char*>(user), key_len)))
+    {
       my_atomic_add32(&number_of_keys_generated,1);
+      mysql_mutex_lock(&LOCK_keys_in_keyring);
```

Comment:
somebody should have implemented scoped locks for MySQL mutexes a long time ago :)